pkg/utils/gardener/shoot.go

```diff
@@ -773,3 +775,81 @@ func GetIPStackForShoot(shoot *gardencorev1beta1.Shoot) string {
 	}
 	return getIPStackForFamilies(ipFamilies)
 }
+
+// CalculateMaxNodesForShoot returns the maximum number of nodes the shoot supports. Function returns nil if there is no limitation.
+func CalculateMaxNodesForShoot(shoot *gardencorev1beta1.Shoot) (*int64, error) {
```

Comment:
Any reason for this being part of the utils package? Do you expect others to require this functionality as well? If not, I guess `pkg/gardenlet/operation/botanist/clusterautoscaler.go` is a better place for it?

Reply:
It was a leftover when I planned to make this field configurable. In this case I had to calculate defaults at different places. That's not required anymore, so I moved the function as you suggested.